browser/profiles/brave_profile_manager.cc

```diff
@@ -117,9 +116,7 @@ void BraveProfileManager::DoFinalInitForServices(Profile* profile,
   DCHECK(status);
   status->UpdateGCMDriverStatus();
 #endif
-  if (base::FeatureList::IsEnabled(brave_news::features::kBraveNewsFeature)) {
-    brave_news::BraveNewsControllerFactory::GetForContext(profile);
-  }
+  brave_news::BraveNewsControllerFactory::GetForContext(profile);
```

Comment:
this is not correct, you should return true from `BrowserContextKeyedServiceFactory::ServiceIsCreatedWithContext`

Reply:
Do you mean this should override `ServiceIsCreatedWithContext` and we should call `brave_news::BraveNewsControllerFactory::GetInstance()` here instead?

Is it just that its more idiomatic, or will there be some behavioral change?

I did notice that there seem to be two methods `GetForContext` and `GetControllerForContext` which do the same thing here.

Reply:
Ah, I see - we already call `GetInstance` in `EnsureBrowserContextKeyedServiceFactoriesBuilt`

Reply:
https://github.com/brave/brave-core/pull/24718